doltpy/cli/dolt.py

```diff
@@ -101,28 +103,71 @@ def __str__(self):
         return f"DoltTable(name: {self.name}, table_hash: {self.table_hash}, rows: {self.rows}, system: {self.system})"
 
 
+@dataclass
 class DoltCommit:
     """
     Represents metadata about a commit, including a ref, timestamp, and author, to make it easier to sort and present
     to the user.
     """
 
-    def __init__(
-        self,
-        ref: str,
-        ts: datetime.datetime,
-        author: str,
-        message: str,
-        merge: Optional[Tuple[str, ...]] = None,
-    ):
-        self.hash = ref
-        self.ts = ts
-        self.author = author
-        self.message = message
-        self.merge = merge
+    ref: str
+    ts: datetime.datetime
+    author: str
+    email: str
+    message: str
+    parent_or_parents: Optional[Union[str, Tuple[str, str]]] = None
 
     def __str__(self):
-        return f"{self.hash}: {self.author} @ {self.ts}, {self.message}"
+        return f"{self.ref}: {self.author} @ {self.ts}, {self.message}"
+
+    def is_merge(self):
+        return isinstance(self.parent_or_parents, tuple)
+
+    def append_merge_parent(self, other_merge_parent: str):
+        if isinstance(self.parent_or_parents, tuple):
+            raise ValueError("Already has a merge parent set")
+        elif not self.parent_or_parents:
+            raise ValueError("No merge parents set")
+        self.parent_or_parents = (self.parent_or_parents, other_merge_parent)
+
+    @classmethod
+    def get_log_table_query(cls):
+        return f"""
+            SELECT
+                dc.`commit_hash`,
+                dca.`parent_hash`,
+                `committer`,
+                `email`,
+                `date`,
+                `message`
+            FROM
+                dolt_commits AS dc
+                LEFT OUTER JOIN dolt_commit_ancestors AS dca
+                    ON dc.commit_hash = dca.commit_hash
+            ORDER BY
+                `date` DESC
+        """
+
+    @classmethod
+    def parse_dolt_log_table(cls, rows: List[dict]) -> OrderedDict:
```

Comment:
not sure if this makes sense here, but if cli row -> Commit is a common transform this might be a bit easier:
```python
def parse_dolt_log_table(cls, rows: List[dict]) -> Dict[str, DoltCommit]:
    commits = defaultdict(list)
    for r in rows:
        ref = row["commit_hash"]
        commits[ref].append(DoltCommit.from_dict(r))
    return commits
```

Reply:
I think we isolated to a single call-site.